In [1]:
from dbfread import DBF
import pandas as pd
from pandas import DataFrame

## Load 10 years of accident data, from 2007 to 2016

In [2]:
dbf07= DBF('accident/accident2007.dbf')
dbf08= DBF('accident/accident2008.dbf')
dbf09= DBF('accident/accident2009.dbf')
dbf10= DBF('accident/accident2010.dbf')
dbf11 = DBF('accident/accident2011.dbf')
dbf12 = DBF('accident/accident2012.dbf')
dbf13 = DBF('accident/accident2013.dbf')
dbf14 = DBF('accident/accident2014.dbf')
dbf15 = DBF('accident/accident2015.dbf')
dbf16 = DBF('accident/accident2016.dbf')
accidents07 = DataFrame(iter(dbf07))
accidents08 = DataFrame(iter(dbf08))
accidents09 = DataFrame(iter(dbf09))
accidents10 = DataFrame(iter(dbf10))
accidents11 = DataFrame(iter(dbf11))
accidents12 = DataFrame(iter(dbf12))
accidents13 = DataFrame(iter(dbf13))
accidents14 = DataFrame(iter(dbf14))
accidents15 = DataFrame(iter(dbf15))
accidents16 = DataFrame(iter(dbf16))


First, we want to combine accidents10 ~ accidents16 to one dataframe. Since not all of the accident data downloaded from the U.S. Department of Transportation have the same features, by using the `jion:inner` option in `pd.concat` function, we can get the intersection of features.

In [3]:
# rename column name in frame07 so that columns names are the same with other frames
accidents07.rename(columns={'latitude': 'LATITUDE', 'longitud': 'LONGITUD'}, inplace=True)

In [4]:
# take a look inside how the accident data file looks like
allaccidents = pd.concat([accidents07,accidents08,accidents09,accidents10,accidents11,accidents12,accidents13,accidents14,accidents15,accidents16], axis=0,join='inner')
pd.set_option('display.max_columns', 100)
allaccidents.head()

STATE  COUNTY  MONTH   DAY  HOUR  MINUTE  VE_TOTAL  PERSONS  PEDS  NHS  \
0    1.0    73.0    1.0   2.0  23.0    15.0       1.0      1.0   0.0  0.0   
1    1.0    19.0    1.0  30.0  13.0     5.0       3.0      4.0   0.0  0.0   
2    1.0     9.0    1.0  14.0  12.0    13.0       2.0      5.0   0.0  0.0   
3    1.0    15.0    1.0   1.0  15.0     0.0       2.0      2.0   0.0  1.0   
4    1.0    51.0    1.0   6.0  19.0    35.0       2.0      5.0   0.0  0.0   

   ROUTE  SP_JUR  HARM_EV  MAN_COLL  REL_ROAD  LGT_COND  WEATHER1  WEATHER2  \
0    4.0     0.0     42.0       0.0       4.0       2.0       1.0       0.0   
1    3.0     0.0     12.0       5.0       1.0       1.0       1.0       0.0   
2    2.0     0.0     12.0       7.0       1.0       1.0       1.0       0.0   
3    2.0     0.0     12.0       4.0       1.0       1.0       1.0       0.0   
4    3.0     0.0     12.0       5.0       1.0       3.0       1.0       0.0   

   NOT_HOUR  NOT_MIN  ARR_HOUR  ARR_MIN  HOSP_HR  HOSP_MN  SCH_BUS  CF1  CF2  \
0      23.0     16.0      23.0     20.0      0.0      0.0      0.0  0.0  0.0   
1      13.0     10.0      13.0     11.0     99.0     99.0      0.0  0.0  0.0   
2      12.0     16.0      12.0     18.0     99.0     99.0      0.0  0.0  0.0   
3      99.0     99.0      15.0      5.0      0.0      0.0      0.0  0.0  0.0   
4      19.0     35.0      19.0     41.0     99.0     99.0      0.0  0.0  0.0   

   CF3  FATALS  DAY_WEEK  DRUNK_DR  ST_CASE    CITY  MILEPT    YEAR  TWAY_ID  \
0  0.0     1.0       3.0       0.0  10001.0     0.0     0.0  2007.0     1493   
1  0.0     2.0       3.0       0.0  10002.0   630.0     0.0  2007.0    SR-68   
2  0.0     1.0       1.0       0.0  10003.0  2330.0     0.0  2007.0  US-SR53   
3  0.0     1.0       2.0       0.0  10004.0     0.0  2468.0  2007.0   US-SR1   
4  0.0     1.0       7.0       0.0  10005.0  2063.0     0.0  2007.0    SR-14   

   TWAY_ID2     RAIL  VE_FORMS  WEATHER   LATITUDE   LONGITUD  
0  00000000  0000000       1.0      1.0  33.455839 -87.017928  
1       270  0000000       3.0      1.0  34.160597 -85.678075  
2  00000000  0000000       2.0      1.0  33.975717 -86.496347  
3      7260  0000000       2.0      1.0  33.853258 -85.915622  
4      I-65  0000000       2.0      1.0  32.488833 -86.408936

In [15]:
accidents11.head()

STATE  ST_CASE  VE_TOTAL  VE_FORMS  PVH_INVL  PEDS  PERNOTMVIT  PERMVIT  \
0      1    10001         1         1         0     0           0        1   
1      1    10002         1         1         0     0           0        1   
2      1    10003         2         2         0     0           0        5   
3      1    10005         1         1         0     0           0        1   
4      1    10006         1         1         0     0           0        2   

   PERSONS  COUNTY  CITY  DAY  MONTH  YEAR  DAY_WEEK  HOUR  MINUTE  NHS  \
0        1       5  1050    4      1  2011         3    16      51    1   
1        1      73  1750    4      1  2011         3    23      40    0   
2        5      97  2100    5      1  2011         4     9      14    0   
3        1      15   100    8      1  2011         7    15      35    0   
4        2      71  2850    8      1  2011         7     6      16    1   

   ROAD_FNC  ROUTE  TWAY_ID TWAY_ID2  MILEPT   LATITUDE   LONGITUD  SP_JUR  \
0         2      2  US-SR 1              700  31.933808 -85.137519       0   
1        16      6     7840                0  33.545008 -86.652594       0   
2        15      6     5884                0  30.707894 -88.173417       0   
3        14      6     5602                0  33.676503 -85.841267       0   
4         2      2  US-SR 2             1505  34.783656 -85.886628       0   

   HARM_EV  MAN_COLL  RELJCT1  RELJCT2  TYP_INT  REL_ROAD  WRK_ZONE  LGT_COND  \
0       34         0        0        1        1         4         0         5   
1       35         0        0        1        1         4         0         3   
2       12         2        0        1        1         1         0         1   
3       34         0        0        1        1         4         0         1   
4       34         0        0        1        1         4         0         3   

   WEATHER1  WEATHER2  WEATHER  SCH_BUS     RAIL  NOT_HOUR  NOT_MIN  ARR_HOUR  \
0         1         0        1        0  0000000        99       99        17   
1         1         0        1        0  0000000        23       46        23   
2         2         0        2        0  0000000        99       99         9   
3         1         0        1        0  0000000        99       99        15   
4         1         0        1        0  0000000        99       99         6   

   ARR_MIN  HOSP_HR  HOSP_MN  CF1  CF2  CF3  FATALS  DRUNK_DR  
0        2       99       99    0    0    0       1         1  
1       50       88       88    0    0    0       1         0  
2       16       99       99    0    0    0       2         1  
3       40       88       88    0    0    0       1         0  
4       16       99       99    0    0    0       1         0

In [6]:
allaccidents.shape

(320874, 42)

The allaccidents table recorded 320874 accidents from 2010-2016, and it has 42 features. Here are the meaning of some of the features according to the `FARS Analytical User’s Manual`.

### Explaination of variables
*VE_TOTAL*: Number of Vehicle in crash <br/>
*VE_FORMS*: Number of Motor Vehicles in Transport (MVIT) <br/>
*PED*: Number of Persons Not in Motor Vehicles <br/>
*NHS*: National Highway System<br/>
*ROUTE*: Route Signing <br/>
*SP_JUR*: Special Jurisdiction <br/>
*HARM_EV*: First Harmful Event<br/>
*TWAY_ID , TWAY_ID2* : Trafficway Identifier <br/>
*MILEPT*: Milepoint <br/>
*SP_JUR*: Special Jurisdiction<br/>
*HARM_EV*: injury or damage producing First Harmful Event <br/> 
*MAN_COLL*:Manner of Collision  <br/> 
*RELJCT1, RELJCT2*: Relation to Junction- Within Interchange Area, Specific Location. <br/>
*TYP_INT*: Type of Intersection <br/>
*REL_ROAD*: Relation to Trafficway <br/>
*LGT_COND*: Light Condition<br/> 
*NOT_HOUR,MIN*: Min, Hour of Notification <br/>
*ARR_HOUR,MIN*: Hour, Min arrival at scene <br/>
*HOSP_HR,MIN*: Hour, Min arrival at hospital <br/>
*CF1, CF2, CF3*:Related Factors- Crash Level, factors related to the crash <br/>
*FATALS*: Fatalities<br/>
*DRUNK_DR*: Number of Drinking Drivers<br/> 
*RAIL*: Rail Grade Crossing Identifier<br/>

For more detailed information, please refer to `FARS Analytical User’s Manual`.

## Select variables and rename variables

Observed from the table above, some of the variables in the table are not very readable. Therefore, in order to make it easier to understand the variables,we renamed some of the variables according to `FARS Analytical User’s Manual`  downloaded from the  `U.S. Department of Transportation`  website.

In [8]:
#combine all vehicle file
allaccidents = accidents07.append([accidents08,accidents09,accidents10,accidents11,accidents12,accidents13,accidents14,accidents15,accidents16])
accidents = allaccidents[['YEAR','ST_CASE','STATE','VE_TOTAL','FATALS','MONTH','DAY_WEEK','HOUR','NHS','LATITUDE','LONGITUD','MAN_COLL','LGT_COND','WEATHER','DRUNK_DR']]
accidents.rename(columns={'ST_CASE':'CASE_NUM','VE_TOTAL':'NUM_VEHICLE','NHS': 'HIGHWAY', 'MAN_COLL': 'COLLISION_TYPE','LGT_COND':'LIGHT_CONDITION','CF1':'CRASH_FACTOR','DRUNK_DR':'DRUNK_DRIVE'}, inplace=True)
accidents['MONTH'] = accidents['MONTH'].map({1.0:'January', 2.0:'February', 3.0: 'March', 4.0:'April', 5.0:'May', 6.0:'June', 7.0:'July', 8.0:'August',9.0: 'September', 10.0:'October', 11.0:'November', 12.0:'December'})
accidents['DAY_WEEK']= accidents['DAY_WEEK'].map({1.0:'Sunday',2.0:'Monday', 3.0:'Tuesday', 4.0: 'Wednesday', 5.0:'Thursday', 6.0:'Friday', 7.0:'Saturday'})
accidents['HIGHWAY'] = accidents['HIGHWAY'].map({1.0:'On',0.0:'Off',9.0:'Unknow'})
accidents['COLLISION_TYPE'] = accidents['COLLISION_TYPE'].map({0.0:'Not Collision',1.0:'Rear-End',2.0:'Head-On',3.0:'Rear-to-Rear',4.0:'Angle',5.0:'Sideswipe, Same Direction',6.0:'Sideswipe, Opposite Direction',7.0:'Sideswipe, Unknown Direction',9.0:'Unknown'})
accidents['LIGHT_CONDITION'] = accidents['LIGHT_CONDITION'].map({1.0:'Daylight',2.0:'Dark' ,3.0:'Dark',5.0:'Dusk',6.0:'Dark',4.0:'Dawn', 7.0:'Other',8.0 :'Not Report', 9.0:'Not Report'})
accidents['WEATHER'] = accidents['WEATHER'].map({0.0:'Normal',1.0:'Clear',2.0:'Rain',3.0:'Sleet',4.0:'Snow',5.0:'Fog, Smog, Smoke',6.0:'Severe Crosswinds',7.0:'Blowing Sand, Soil, Dirt',8.0:'Other',10.0:'Cloudy',11.0:'Blowing Snow',12.0:'Freezing Rain or Drizzle',98.0:'Unknown',99.0:'Unknown'})
accidents.head()

/Users/aitingkuang/anaconda/lib/python3.6/site-packages/pandas/core/frame.py:2844: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  **kwargs)
/Users/aitingkuang/anaconda/lib/python3.6/site-packages/ipykernel_launcher.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """
/Users/aitingkuang/anaconda/lib/python3.6/site-packages/ipykernel_launcher.py:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexin

YEAR  CASE_NUM  STATE  NUM_VEHICLE  FATALS    MONTH  DAY_WEEK  HOUR  \
0  2007.0   10001.0    1.0          1.0     1.0  January   Tuesday  23.0   
1  2007.0   10002.0    1.0          3.0     2.0  January   Tuesday  13.0   
2  2007.0   10003.0    1.0          2.0     1.0  January    Sunday  12.0   
3  2007.0   10004.0    1.0          2.0     1.0  January    Monday  15.0   
4  2007.0   10005.0    1.0          2.0     1.0  January  Saturday  19.0   

  HIGHWAY   LATITUDE   LONGITUD                COLLISION_TYPE LIGHT_CONDITION  \
0     Off  33.455839 -87.017928                 Not Collision            Dark   
1     Off  34.160597 -85.678075     Sideswipe, Same Direction        Daylight   
2     Off  33.975717 -86.496347  Sideswipe, Unknown Direction        Daylight   
3      On  33.853258 -85.915622                         Angle        Daylight   
4     Off  32.488833 -86.408936     Sideswipe, Same Direction            Dark   

  WEATHER  DRUNK_DRIVE  
0   Clear          0.0  
1   Clear          0.0  
2   Clear          0.0  
3   Clear          0.0  
4   Clear          0.0

### Load vehicle data file which contains mortality rate

We also want to study the mortality rate of fatal accidents. The data element “Fatalities in Vehicle” in the Vehicle data file from the `U.S. Department of Transportation` website provides the number of deaths in a vehicle.

In [9]:
vdbf07= DBF('vehicle_deaths/vehicle2007.dbf')
vdbf08= DBF('vehicle_deaths/vehicle2008.dbf')
vdbf09= DBF('vehicle_deaths/vehicle2009.dbf')
vdbf10= DBF('vehicle_deaths/vehicle2010.dbf')
vdbf11= DBF('vehicle_deaths/vehicle2011.dbf')
vdbf12= DBF('vehicle_deaths/vehicle2012.dbf')
vdbf13= DBF('vehicle_deaths/vehicle2013.dbf')
vdbf14= DBF('vehicle_deaths/vehicle2014.dbf')
# vdbf15= DBF('vehicle_deaths/vehicle2015.csv')
vdbf16= DBF('vehicle_deaths/vehicle2016.dbf')
vehicle07 = DataFrame(iter(vdbf07))
vehicle08 = DataFrame(iter(vdbf08))
vehicle09 = DataFrame(iter(vdbf09))
vehicle10 = DataFrame(iter(vdbf10))
vehicle11 = DataFrame(iter(vdbf11))
vehicle12 = DataFrame(iter(vdbf12))
vehicle13 = DataFrame(iter(vdbf13))
vehicle14 = DataFrame(iter(vdbf14))
# vehicle15 = pd.read_csv('vehicle_deaths/vehicle2015.csv')
vehicle16 = DataFrame(iter(vdbf16))

In [10]:
vehicle07['YEAR']='2007.0'
vehicle08['YEAR']='2008.0'
vehicle09['YEAR']='2009.0'
vehicle10['YEAR']='2010.0'
vehicle11['YEAR']='2011.0'
vehicle12['YEAR']='2012.0'
vehicle13['YEAR']='2013.0'
vehicle14['YEAR']='2014.0'
# vehicle15['YEAR']='2015.0'
vehicle16['YEAR']='2016.0'
allvehicle = pd.concat([vehicle07,vehicle08,vehicle09,vehicle10,vehicle11,vehicle12,vehicle13,vehicle14,vehicle16], axis=0,join='outer')
allvehicle.head()

ACC_TYPE  AVOID  AXLES  BODY_TYP  BUS_USE CARBUR  CARGO_BT  CDL_STAT  \
0       NaN    1.0    0.0       4.0      0.0    NaN       0.0       0.0   
1       NaN    0.0    0.0       4.0      0.0    NaN       0.0       0.0   
2       NaN    0.0    0.0       4.0      0.0    NaN       0.0       0.0   
3       NaN    8.0    0.0      31.0      0.0    NaN       0.0       0.0   
4       NaN    8.0    0.0      15.0      0.0    NaN       0.0       0.0   

  CYLINDER  DAY  DEATHS  DEFORMED DISPLACE  DR_CF1  DR_CF2  DR_CF3  DR_CF4  \
0      NaN  NaN     1.0       6.0      NaN    44.0    28.0     0.0     0.0   
1      NaN  NaN     0.0       6.0      NaN    38.0     0.0     0.0     0.0   
2      NaN  NaN     2.0       6.0      NaN     0.0     0.0     0.0     0.0   
3      NaN  NaN     0.0       6.0      NaN     0.0     0.0     0.0     0.0   
4      NaN  NaN     0.0       2.0      NaN    90.0    53.0     0.0     0.0   

   DR_DRINK  DR_HGT  DR_PRES  DR_SF1  DR_SF2  DR_SF3  DR_SF4  DR_WGT   DR_ZIP  \
0       0.0    68.0      1.0     NaN     NaN     NaN     NaN   175.0  35117.0   
1       0.0    63.0      1.0     NaN     NaN     NaN     NaN   115.0  35960.0   
2       0.0    71.0      1.0     NaN     NaN     NaN     NaN   185.0  35983.0   
3       0.0    72.0      1.0     NaN     NaN     NaN     NaN   195.0  35983.0   
4       0.0    70.0      1.0     NaN     NaN     NaN     NaN   180.0  36203.0   

   D_VISION1  D_VISION2  D_VISION3  EMER_USE  FIRE_EXP  FIRST_MO  FIRST_YR  \
0        NaN        NaN        NaN       0.0       0.0       8.0    2006.0   
1        NaN        NaN        NaN       0.0       0.0       0.0       0.0   
2        NaN        NaN        NaN       0.0       0.0       0.0       0.0   
3        NaN        NaN        NaN       0.0       0.0       0.0       0.0   
4        NaN        NaN        NaN       0.0       0.0       0.0       0.0   

  FLDCD_TR FUELCODE  GVWR  HARM_EV  HAZ_CNO  HAZ_ID  HAZ_INV  HAZ_PLAC  \
0        *      NaN   0.0     42.0      0.0     0.0      1.0       0.0   
1        *      NaN   0.0     12.0      0.0     0.0      1.0       0.0   
2        *      NaN   0.0     12.0      0.0     0.0      1.0       0.0   
3        G      NaN   0.0     12.0      0.0     0.0      1.0       0.0   
4        *      NaN   0.0     12.0      0.0     0.0      1.0       0.0   

   HAZ_REL  HIT_RUN  HOUR  IMPACT1  IMPACT2  IMPACTS  J_KNIFE  LAST_MO  \
0      0.0      0.0   NaN      9.0      9.0      1.0      0.0      9.0   
1      0.0      0.0   NaN     12.0     12.0      1.0      0.0      0.0   
2      0.0      0.0   NaN      7.0      3.0      2.0      0.0      0.0   
3      0.0      0.0   NaN     12.0     12.0      1.0      0.0      0.0   
4      0.0      1.0   NaN      5.0      5.0      1.0      1.0      0.0   

   LAST_YR   ...    TRLR2VIN  TRLR3VIN  UNDERIDE  UNITTYPE  VALIGN  VEH_CF1  \
0   2006.0   ...         NaN       NaN       0.0       1.0     NaN     99.0   
1      0.0   ...         NaN       NaN       0.0       1.0     NaN      0.0   
2      0.0   ...         NaN       NaN       0.0       1.0     NaN      0.0   
3      0.0   ...         NaN       NaN       0.0       1.0     NaN      0.0   
4      0.0   ...         NaN       NaN       0.0       1.0     NaN      1.0   

   VEH_CF2  VEH_MAN  VEH_NO  VEH_SC1 VEH_SC2 VE_FORMS           VIN  VINA_MOD  \
0     99.0     17.0     1.0      NaN     NaN      1.0  JHMCB7657NC0       ALE   
1      0.0      3.0     1.0      NaN     NaN      3.0  JN1HJ01P7LT4       MAX   
2      0.0      1.0     2.0      NaN     NaN      3.0  1G6CD53BXP42       DFW   
3      0.0      1.0     3.0      NaN     NaN      3.0  1FTPW12524KB       F1S   
4     31.0      5.0     1.0      NaN     NaN      2.0  1D4HD58266F1       DUR   

  VINMAKE  VINMODYR  VINTYPE  VIN_1  VIN_10  VIN_11  VIN_12  VIN_2  VIN_3  \
0     NaN       NaN      NaN      J       N       C       0      H      M   
1     NaN       NaN      NaN      J       L       T       4      N      1   
2     NaN       NaN      NaN      1       P 

In [11]:
#
vehicles = allvehicle[['YEAR','ST_CASE','HIT_RUN','TRAV_SP','ROLLOVER','FIRE_EXP','SPEEDREL','DEATHS']]
vehicles.rename(columns={'ST_CASE':'CASE_NUM','TRAV_SP':'SPEED','FIRE_EXP': 'FIRE','SPEEDREL':'SPEEDING'}, inplace=True)
vehicles.head(300)

/Users/aitingkuang/anaconda/lib/python3.6/site-packages/pandas/core/frame.py:2844: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  **kwargs)


YEAR  CASE_NUM  HIT_RUN  SPEED  ROLLOVER  FIRE  SPEEDING  DEATHS
0    2007.0   10001.0      0.0   65.0       0.0   0.0       NaN     1.0
1    2007.0   10002.0      0.0   10.0       0.0   0.0       NaN     0.0
2    2007.0   10002.0      0.0   55.0       0.0   0.0       NaN     2.0
3    2007.0   10002.0      0.0   55.0       0.0   0.0       NaN     0.0
4    2007.0   10003.0      1.0   55.0       0.0   0.0       NaN     0.0
5    2007.0   10003.0      1.0   50.0       2.0   0.0       NaN     1.0
6    2007.0   10004.0      0.0   20.0       0.0   0.0       NaN     1.0
7    2007.0   10004.0      0.0   62.0       0.0   0.0       NaN     0.0
8    2007.0   10005.0      0.0   15.0       0.0   0.0       NaN     1.0
9    2007.0   10005.0      0.0   50.0       0.0   0.0       NaN     0.0
10   2007.0   10006.0      0.0   75.0       1.0   0.0       NaN     1.0
11   2007.0   10007.0      0.0   55.0       0.0   0.0       NaN     1.0
12   2007.0   10007.0      0.0   35.0       0.0   0.0       NaN     0.0
13   2007.0   10008.0      0.0   99.0       0.0   0.0       NaN     0.0
14   2007.0   10008.0      0.0   99.0       0.0   0.0       NaN     1.0
15   2007.0   10009.0      0.0   55.0       0.0   0.0       NaN     1.0
16   2007.0   10009.0      0.0   50.0       0.0   0.0       NaN     0.0
17   2007.0   10010.0      0.0   55.0       0.0   0.0       NaN     1.0
18   2007.0   10010.0      0.0   50.0       0.0   0.0       NaN     0.0
19   2007.0   10011.0      0.0   80.0       1.0   0.0       NaN     1.0
20   2007.0   10012.0      0.0   70.0       0.0   0.0       NaN     1.0
21   2007.0   10012.0      0.0   70.0       0.0   0.0       NaN     0.0
22   2007.0   10012.0      0.0   70.0       0.0   0.0       NaN     0.0
23   2007.0   10013.0      0.0   50.0       2.0   0.0       NaN     1.0
24   2007.0   10014.0      0.0   70.0       2.0   0.0       NaN     1.0
25   2007.0   10015.0      0.0   45.0       0.0   0.0       NaN     0.0
26   2007.0   10015.0      0.0   45.0       0.0   0.0       NaN     0.0
27   2007.0   10016.0      0.0   76.0       2.0   0.0       NaN     2.0
28   2007.0   10017.0      0.0   70.0       0.0   0.0       NaN     1.0
29   2007.0   10018.0      0.0   70.0       0.0   0.0       NaN     1.0
..      ...       ...      ...    ...       ...   ...       ...     ...
270  2007.0   10190.0      0.0   60.0       2.0   0.0       NaN     1.0
271  2007.0   10191.0      0.0   60.0       0.0   0.0       NaN     0.0
272  2007.0   10191.0      0.0   25.0       0.0   0.0       NaN     1.0
273  2007.0   10192.0      0.0   80.0       1.0   0.0       NaN     1.0
274  2007.0   10193.0      0.0   55.0       0.0   0.0       NaN     1.0
275  2007.0   10193.0      0.0   55.0       0.0   0.0       NaN     0.0
276  2007.0   10194.0      0.0   99.0       0.0   0.0       NaN     1.0
277  2007.0   10194.0      0.0   99.0       0.0   0.0       NaN     0.0
278  2007.0   10195.0      0.0   40.0       2.0   0.0       NaN     1.0
279  2007.0   10196.0      0.0   60.0       0.0   0.0       NaN     1.0
280  2007.0   10197.0      0.0   70.0       0.0   0.0       NaN     1.0
281  2007.0   10197.0      0.0   10.0       0.0   0.0       NaN     0.0
282  2007.0   10198.0      0.0   80.0       2.0   0.0       NaN     1.0
283  2007.0   10199.0      0.0   60.0       0.0   0.0       NaN     0.0
284  2007.0   10199.0      0.0   60.0       0.0   0.0       NaN     2.0
285  2007.0   10200.0      0.0   10.0       0.0   0.0       NaN     0.0
286  2007.0   10200.0      0.0   40.0       0.0   0.0       NaN     1.0
287  2007.0   10201.0      0.0   65.0       1.0   0.0       NaN     1.0
288  2007.0   10202.0      0.0   45.0       0.0   0.0       NaN     1.0
289  2007.0   10202.0      0.0   45.0       0.0   0.0       NaN     0.0
290  2007.0   10203.0      0.0   52.0       0.0   0.0       NaN     1.0
291  2007.0   10204.0      0.0   99.0       1.0   0.0       NaN     1.0
292  2007.0   10204.0      0.0   25.0       0.0   0.0       NaN     0.0
293  2007.0   10205.0      0.0   65.0       2.0